arangod/RestHandler/RestDocumentHandler.cpp

```diff
@@ -316,31 +317,34 @@ RestStatus RestDocumentHandler::readSingleDocument(bool generateBody) {
   }
 
   return waitForFuture(
-      _activeTrx->documentAsync(collection, search, options).thenValue([=, buffer(std::move(buffer))](OperationResult opRes) {
-        Result res = _activeTrx->finish(opRes.result);
-
-        if (!opRes.ok()) {
-          generateTransactionError(collection, opRes, key, ifRid);
-          return;
-        }
-
-        if (!res.ok()) {
-          generateTransactionError(collection, OperationResult(res, options), key, ifRid);
-          return;
-        }
-
-        if (ifNoneRid.isSet()) {
-          RevisionId const rid = RevisionId::fromSlice(opRes.slice());
-          if (ifNoneRid == rid) {
-            generateNotModified(rid);
-            return;
-          }
-        }
-
-        // use default options
-        generateDocument(opRes.slice(), generateBody,
-                         _activeTrx->transactionContextPtr()->getVPackOptions());
-      }));
+      _activeTrx->documentAsync(collection, search, options)
```

Comment:
Wondering if this is actually necessary.
This is a read-only operation, so we don't need to commit anything locally. The transaction can just be aborted. It still makes sense if this can be called from a mixed (RW) transaction or if this actually does communication with followers.
If it is just a local operation, we don't need to make it async by all means.

Reply:
I think in the case that it does everything locally then it's still basically fine, right? It adds one more callback to the future chain, but since the sync version executes the async version and waits on the future, there's not a whole lot of difference. Unless it specifically makes a network call, then all the callbacks should still happen in the same thread. It's only put back in the scheduler if it hits the network, right?